src/libstore/machines.hh

```diff
@@ -1,48 +1,67 @@
 #pragma once
 ///@file
 
+#include "build-capability.hh"
 #include "types.hh"
 
 namespace nix {
 
 class Store;
 
+struct SchedulableCapability;
+
 struct Machine {
 
     const std::string storeUri;
-    const std::set<std::string> systemTypes;
+
     const std::string sshKey;
-    const unsigned int maxJobs;
```

Comment:
Actually removing this is probably wrong, because it can be used to represent a limit on a single resource, such as an x86_64 + i686 build capability, or the resources of VMs (+ non-VM capabilities) such as an aarch64-darwin host with an aarch64-linux build VM.
So yes, we probably want maxJobs in both the machine and in the capabilities.

This limit may be crude, but the same idea applies when it's, say, a (cpu, memory) tuple instead of a single number of jobs; that's an orthogonal issue.